### ISSUE TITLE
"by" option not working hvplot

### ISSUE THREAD

**Issue by TylerTCF** [pyviz/hvplot#219]
The code below is not functional for me after upgrading hvplot, specifically for pandas.  I have implemented the following versions of dependencies:

Python=3.6.8
Hvplot=0.4.0
Bokeh=1.2.0
Holoviews=1.11.3
PyViz Comms=0.7.2

```
import pandas as pd
import holoviews as hv
import hvplot.pandas
hv.extension('bokeh')
d={'Col A':['Mike', 'Bill', 'Sue', 'Lauren'], 'Col B':['House', 'House', 'Apartment', 'Apartment'], 'Col C':[9,1,2, 3]}
a=pd.DataFrame(d)
a.hvplot('Col A', 'Col C', by='Col B')
```
Output is:
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
~\Anaconda3\lib\site-packages\IPython\core\formatters.py in __call__(self, obj, include, exclude)
    968 
    969             if method is not None:
--> 970                 return method(include=include, exclude=exclude)
    971             return None
    972         else:

~\Anaconda3\lib\site-packages\holoviews\core\dimension.py in _repr_mimebundle_(self, include, exclude)
   1438         combined and returned.
   1439         """
-> 1440         return Store.render(self)
   1441 
   1442 

~\Anaconda3\lib\site-packages\holoviews\core\options.py in render(cls, obj)
   1514         data, metadata = {}, {}
   1515         for hook in hooks:
-> 1516             ret = hook(obj)
   1517             if ret is None:
   1518                 continue

~\Anaconda3\lib\site-packages\holoviews\ipython\display_hooks.py in pprint_display(obj)
    275     if not ip.display_formatter.formatters['text/plain'].pprint:
    276         return None
--> 277     return display(obj, raw_output=True)
    278 
    279 

~\Anaconda3\lib\site-packages\holoviews\ipython\display_hooks.py in display(obj, raw_output, **kwargs)
    245     elif isinstance(obj, (CompositeOverlay, ViewableElement)):
    246         with option_state(obj):
--> 247             output = element_display(obj)
    248     elif isinstance(obj, (Layout, NdLayout, AdjointLayout)):
    249         with option_state(obj):

~\Anaconda3\lib\site-packages\holoviews\ipython\display_hooks.py in wrapped(element)
    140         try:
    141             max_frames = OutputSettings.options['max_frames']
--> 142             mimebundle = fn(element, max_frames=max_frames)
    143             if mimebundle is None:
    144                 return {}, {}

~\Anaconda3\lib\site-packages\holoviews\ipython\display_hooks.py in element_display(element, max_frames)
    186         return None
    187 
--> 188     return render(element)
    189 
    190 

~\Anaconda3\lib\site-packages\holoviews\ipython\display_hooks.py in render(obj, **kwargs)
     63         renderer = renderer.instance(fig='png')
     64 
---> 65     return renderer.components(obj, **kwargs)
     66 
     67 

~\Anaconda3\lib\site-packages\holoviews\plotting\bokeh\renderer.py in components(self, obj, fmt, comm, **kwargs)
    247         # Bokeh has to handle comms directly in <0.12.15
    248         comm = False if bokeh_version < '0.12.15' else comm
--> 249         return super(BokehRenderer, self).components(obj,fmt, comm, **kwargs)
    250 
    251 

~\Anaconda3\lib\site-packages\holoviews\plotting\renderer.py in components(self, obj, fmt, comm, **kwargs)
    319             plot = obj
    320         else:
--> 321             plot, fmt = self._validate(obj, fmt)
    322 
    323         widget_id = None

~\Anaconda3\lib\site-packages\holoviews\plotting\renderer.py in _validate(self, obj, fmt, **kwargs)
    218         if isinstance(obj, tuple(self.widgets.values())):
    219             return obj, 'html'
--> 220         plot = self.get_plot(obj, renderer=self, **kwargs)
    221 
    222         fig_formats = self.mode_formats['fig'][self.mode]

~\Anaconda3\lib\site-packages\holoviews\plotting\bokeh\renderer.py in get_plot(self_or_cls, obj, doc, renderer, **kwargs)
    132             curdoc().theme = self_or_cls.theme
    133         doc.theme = self_or_cls.theme
--> 134         plot = super(BokehRenderer, self_or_cls).get_plot(obj, renderer, **kwargs)
    135         plot.document = doc
    136         return plot

~\Anaconda3\lib\site-packages\holoviews\plotting\renderer.py in get_plot(self_or_cls, obj, renderer, **kwargs)
    205             init_key = tuple(v if d is None else d for v, d in
    206                              zip(plot.keys[0], defaults))
--> 207             plot.update(init_key)
    208         else:
    209             plot = obj

~\Anaconda3\lib\site-packages\holoviews\plotting\plot.py in update(self, key)
    593     def update(self, key):
    594         if len(self) == 1 and ((key == 0) or (key == self.keys[0])) and not self.drawn:
--> 595             return self.initialize_plot()
    596         item = self.__getitem__(key)
    597         self.traverse(lambda x: setattr(x, '_updated', True))

~\Anaconda3\lib\site-packages\holoviews\plotting\bokeh\element.py in initialize_plot(self, ranges, plot, plots)
   1888             self.handles['plot'] = Tabs(tabs=panels)
   1889         elif not self.overlaid:
-> 1890             self._process_legend()
   1891             self._set_active_tools(plot)
   1892         self.drawn = True

~\Anaconda3\lib\site-packages\holoviews\plotting\bokeh\element.py in _process_legend(self)
   1761         elif pos in ['above', 'below', 'right', 'left']:
   1762             plot.legend.pop(plot.legend.index(legend))
-> 1763             legend.plot = None
   1764             legend.location = self.legend_offset
   1765             plot.add_layout(legend, pos)

~\Anaconda3\lib\site-packages\bokeh\core\has_props.py in __setattr__(self, name, value)
    286 
    287             raise AttributeError("unexpected attribute '%s' to %s, %s attributes are %s" %
--> 288                 (name, self.__class__.__name__, text, nice_join(matches)))
    289 
    290     def __str__(self):

AttributeError: unexpected attribute 'plot' to Legend, possible attributes are background_fill_alpha, background_fill_color, border_line_alpha, border_line_cap, border_line_color, border_line_dash, border_line_dash_offset, border_line_join, border_line_width, click_policy, glyph_height, glyph_width, inactive_fill_alpha, inactive_fill_color, items, js_event_callbacks, js_property_callbacks, label_height, label_standoff, label_text_align, label_text_alpha, label_text_baseline, label_text_color, label_text_font, label_text_font_size, label_text_font_style, label_text_line_height, label_width, level, location, margin, name, orientation, padding, spacing, subscribed_events, tags, title, title_standoff, title_text_align, title_text_alpha, title_text_baseline, title_text_color, title_text_font, title_text_font_size, title_text_font_style, title_text_line_height or visible

:NdOverlay   [Col B]
   :Curve   [Col A]   (Col C)

Thanks!

**Comment by poplarShift** [comment]
Works for me, you could try upgrading to the latest holoviews.

**Comment by TylerTCF** [comment]
Fixed by updating holoviews to 1.12.3.  Thanks!
